public/js/subcircuit.js

```diff
@@ -290,6 +290,17 @@ SubCircuit.prototype.reset = function() {
 
 SubCircuit.prototype.click = function() {
     // this.id=prompt();
+    for(let i = 0; i < this.localScope["Button"].length; i++){
+        if (simulationArea.mouseDownX === this.x + this.localScope["Button"][i].subcircuitMetadata.x 
+            && simulationArea.mouseDownY === this.y + this.localScope["Button"][i].subcircuitMetadata.y){
+                this.lastClickedElement = i;
+                this.localScope["Button"][i].wasClicked = true;
+            }
+    }    
+}
```

Comment:
You cannot check if the mouse is hovering with by just checking if x and y match. You will need to do it similar to `CircuitElement.prototype.isHover()`

Reply:
Why hardcode with Button only? other elements can be clicked as well.